docs/system_overview/ssl_and_tls.md

```diff
@@ -0,0 +1,148 @@
+
+To create the chain file for irods.example.org:
+
+~~~
+irods@hostname:~/ $ cat irods.crt PositiveSSLCA2.crt AddTrustExternalCARoot.crt > chain.pem
+~~~
+
+### Generate OpenSSL parameters
+
+Generate some Diffie-Hellman parameters for OpenSSL:
+
+~~~
+irods@hostname:~/ $ openssl dhparam -2 -out dhparams.pem 2048
+~~~
+
+### Place files within accessible area
+
+Put the dhparams.pem, server.key and chain.pem files somewhere that the iRODS server can access them (e.g. in /etc/irods).  Make sure that the irods unix user can read the files (although you also want to make sure that the key file is only readable by the irods user).
+
+### Configure iRODS server for TLS
+
+The server expects to have the following properties set in `server_config.json` when loading configuration at startup:
+
+~~~
+"tls": {
+    "certificate_chain_file": "/etc/irods/chain.pem",
+    "certificate_key_file": "/etc/irods/server.key",
+    "dh_params_file": "/etc/irods/dhparams.pem"
+}
+~~~
+
+Here, we define what each of these configuration options means:
+
+certificate_chain_file
+:       The file containing the server's certificate chain. The certificates must be in PEM format and must be sorted starting with the subject's certificate (actual client or server certificate), followed by intermediate CA certificates if applicable, and ending at the highest level (root) CA.
+
+certificate_key_file
+:       Private key corresponding to the server's certificate in the certificate chain file.
+
+dh_params_file
+:       The Diffie-Hellman parameter file location.
+
+
+In order for the configuration to take effect, the iRODS server configuration must be reloaded. This can be done by sending a `SIGHUP` to the main iRODS server process:
+
+~~~
+irods@hostname:~/ $ kill -HUP $(cat /var/run/irods/irods-server.pid)
+~~~
+
+## Client SSL Setup
+
+The client may or may not require configuration at the SSL level, but there are a few parameters that can be set via `irods_environment.json` properties to customize the client SSL interaction if necessary. In many cases, if the server's certificate comes from a common CA, your system might already be configured to accept certificates from that CA, and you will not have to adjust the client configuration at all. For example, on Debian-based systems, the `/etc/ssl/certs` directory is used as a repository for system trusted certificates installed via an Ubuntu package. Many of the commercial certificate vendors such as VeriSign and AddTrust have their certificates already installed.
+
+### Server Verification Settings
+
+Server verification can be turned off using the irods_ssl_verify_server `irods_environment.json` property. If this variable is set to 'none', then any certificate (or none) is accepted by the client. This means that your connection will be encrypted, but you cannot be sure to what server (i.e. there is no server authentication). For that reason, this mode is discouraged.
```

Comment:
We should highlight the property too - `irods_ssl_verify_server`, but we can do in a later PR since there are so many cases throughout the docs like this.

Reply:
I went ahead and did it anyway for this case.